lib/api/privilegeLevelLogic.ts

```diff
@@ -0,0 +1,81 @@
+import { PrismaClient, PrivilegeLevel } from '@prisma/client';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { getSession } from 'next-auth/react';
+import privilegeLevel from '../../pages/api/user/[email]/privilegeLevel';
+import { LOWEST_AUTHORIZED_PRIVILEGE_LEVEL } from '../constants';
+import { BadRequestError } from '../utils/errors/badRequestError';
+import {
+  getPrivilegeLevelFromSession,
+  privilegeLevelCompareTo,
+} from '../utils/privilegeLevelUtils';
+
+//TYPES
+export type Context = {
+  prisma: PrismaClient;
+};
+
+type RequestParameters = {
```

Comment:
what about we create a folder for types??? @rymaju and I created /typings for a .d.ts file that we needed to create to override the type for Session (so that it can include the privilege level) 

Could we use the /typings folder to store all future types? The only types from privilgeLevelLogic.ts are Context and RequestParameters which are pretty specific to the privilegeLevel endpoint so maybe they should go in /typings/privilegeLevelTypes.ts?

For the future, when we have common types maybe they should go in /typings/commonTypes.ts (or something? that's not a cute name)

Reply:
RequestParameters is a DTO so it should go in /lib/api. If you don't expect Context to be used outside privilegeLevel files then keep it in this. If you envision it being used for more, you can make it a common type. If you're not sure about this, I would just leave it in this file and we can make it a common type if necessary

Reply:
What makes something a DTO? My understanding is that it is an object that encapsulates data from one application layer to another? Can a type be a DTO?

Reply:
a DTO is an object that abstracts the data layer. Aka you have this object and its methods are used to retrieve data. So when the database switches you don't have to change prisma calls to maria db calls everywhere. You only have to change it in the DTO's methods. Think of it an interface of the database/table in the database, it just has an  implementation ;)